complex/src/scalar.rs

```diff
@@ -0,0 +1,252 @@
+
+use traits::{Num, FromPrimitive, Float};
+use std::ops::Neg;
+
+use Complex;
+
+pub trait Scalar: Num + Copy + Neg<Output = Self> {
+    /// Associated Real type
+    type Real;
+    /// Associated Complex type
+    type Complex;
```

Comment:
Do we really need this flexibility?  Perhaps we should just use `Complex<Self::Real>` where needed?  That's only for `powc`, so I wonder if that's worth its weight here at all.